src/braket/aws/aws_session.py

```diff
@@ -115,6 +118,12 @@ def ecr_client(self):
             self._ecr = self.boto_session.client("ecr", region_name=self.region)
         return self._ecr
 
+    @property
+    def pricing_client(self):
+        if not self._pricing:
+            self._pricing = self.boto_session.client("pricing", region_name="us-east-1")
```

Comment:
Could you please validate whether this works for a user IAM role which only has the AmazonBraketFullAccess permissions attached ?

Reply:
Good thought. No, for the tracker to work, the IAM role needs pricing:GetProducts, which AmazonBraketFullAccess does not have.